src/ReferenceList.php

```diff
@@ -286,4 +263,33 @@ public function equals( $target ) {
 		       && $this->getValueHash() === $target->getValueHash();
 	}
 
+	/**
+	 * @see Countable::count
+	 *
+	 * @return int
+	 */
+	public function count() {
+		return count( $this->references );
+	}
+
+	/**
+	 * @see IteratorAggregate::getIterator
+	 *
+	 * @since 5.0
+	 *
+	 * @return Traversable
+	 */
+	public function getIterator() {
```

Comment:
Intention is that we don't break loops that iterate over a `ReferenceList`. `foreach ( $referenceList as $reference ) {` should still work with this.